t/16-utils-git.t

```diff
@@ -0,0 +1,44 @@
+# Copyright SUSE LLC
+# SPDX-License-Identifier: GPL-2.0-or-later
+
+use Test::Most;
+use Test::Warnings ':report_warnings';
+use Test::MockModule;
+use Test::MockObject;
+use FindBin;
+use lib "$FindBin::Bin/lib", "$FindBin::Bin/../external/os-autoinst-common/lib";
+use OpenQA::Test::TimeLimit '10';
+use OpenQA::Git;
+
+my $mock = Test::MockModule->new('OpenQA::Utils');
+$mock->redefine('run_cmd_with_log_return_error', sub {
```

Comment:
Have a look at t/16-utils-runcmd.t
`run_cmd_with_log_return_error` is an imported function, so it is already defined in OpenQA::Git when you create the OpenQA::Util mock.
Just do it like in t/16-utils-runcmd.t

Maybe the tests from there should be moved to your new test?
It would be a bit weird to have them in separate files.